source/developers-guide/customer-streams-extension/index.md

```diff
@@ -0,0 +1,280 @@
+    {
+        /** @var \Shopware_Controllers_Backend_Customer $subject */
+        $subject = $args->getSubject();
+
+        $subject->View()->addTemplateDir(__DIR__ . '/Resources/views');
+
+        $subject->View()->extendsTemplate('backend/customer/swag_customer_stream_extension.js');
+    }
+}
+```
+
+The extended `swag_customer_stream_extension.js` contains all overrides for the backend module:
+
+```
+// {block name="backend/customer/view/customer_stream/condition_panel" }
+
+// {$smarty.block.parent}
+
+Ext.define('Shopware.apps.Customer.SwagCustomerStreamExtension', {
+    override: 'Shopware.apps.Customer.view.customer_stream.ConditionPanel',
+
+    registerHandlers: function() {
+        var me = this,
+            //fetch original handlers
+            handlers = me.callParent(arguments);
+
+        //push own handler into
+        handlers.push(Ext.create('Shopware.apps.Customer.swag_customer_stream_extension.ActiveCondition'));
+
+        //return modified handlers array
+        return handlers;
+    }
+});
+
+
+//definition of you own condition
+Ext.define('Shopware.apps.Customer.swag_customer_stream_extension.ActiveCondition', {
+
+    getLabel: function() {
+        return 'My active condition';
+    },
+
+    supports: function(conditionClass) {
+        return (conditionClass == 'SwagCustomerSearchExtension\\Bundle\\CustomerSearchBundle\\ActiveCondition');
+    },
+
+    create: function(callback) {
+        callback(this._create());
+    },
+
+    load: function(conditionClass, items, callback) {
+        callback(this._create());
+    },
+
+    _create: function() {
+        return {
+            title: this.getLabel(),
+            conditionClass: 'SwagCustomerSearchExtension\\Bundle\\CustomerSearchBundle\\ActiveCondition',
+            items: [{
+                xtype: 'checkbox',
+                name: 'active',
+                boxLabel: 'Activate for active customers, deactivate for inactive customers',
+                inputValue: true,
+                uncheckedValue: false
+            }]
+        };
+    }
+});
+
+// {/block}
+```
+
+The first part hooks into the customer stream condition panel and registers the plugin condition. The second part contains the whole logic to handle the condition for load and create actions. 
+Create and load have to return an object with the following data:
+* `title` - Used for the panel title
+* `conditionClass` - Used for class generation - Aside, used for singleton detection
+* `items` - Contains a list of parameters, which used for __construct call
+
+## Search Indexing
+The `CustomerSearchBundleDBAL` uses an aggregated table in order to be able to filter and sort as much as possible on large data sets. This table is generated over the `Shopware\Bundle\CustomerSearchBundleDBAL\Indexing\SearchIndexer` class.  If a plugin wants to filter and sort additional aggregated data, it can hook into the indexing process to collect additional data.
```

Comment:
The `CustomerSearchBundleDBAL` is based on an aggregated table which allows fast filtering and sorting, even on  large data sets. This table is generated by the `Shopware\Bundle\CustomerSearchBundleDBAL\Indexing\SearchIndexer` class.  If a plugin wants to filter and sort additional aggregated data, it can hook into the indexing process to collect additional data.

*** FRAGE ***
Ist hier wirklich Hook gemeint? Also muss man tatsächlich einen Hook benutzen oder soll mit Hook nur gesagt werden das dies die richtige Stelle ist? Nicht das die Leute später Hooks nutzen obwohl es einen schöneren Weg gibt.